json-mode-social-determinants-of-health/SDOH-Json-mode.ipynb

```diff
@@ -0,0 +1,593 @@
+       "0             False  \n",
+       "1             False  \n",
+       "2             False  \n",
+       "3             False  \n",
+       "4              True  "
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Total latency: 11.91s\n",
+    "\n",
+    "model = \"mixtral-8x7b-32768\"\n",
+    "\n",
+    "patients_data = []\n",
+    "# Loop through each patient clinical note, extract structured SDOH and compile a list of json objects\n",
+    "for file_name in text_files:\n",
+    "    with open(os.path.join(folder_path, file_name), 'r') as file:\n",
+    "        clinical_note = file.read()\n",
+    "        user_prompt = user_prompt_template.format(clinical_note=clinical_note)\n",
+    "        social_determinants_json = extract_sdoh_json(system_prompt,user_prompt,model)\n",
+    "        social_determinants_json['mrn'] = file_name[:-4] # The name of the file is the patient's MRN \n",
+    "        patients_data.append(social_determinants_json)\n",
+    "\n",
+    "# Flatten the results into a dataframe\n",
+    "flattened_data = []\n",
+    "for patient in patients_data:\n",
+    "    flattened_data.append({'mrn': patient['mrn'],\n",
+    "                           'employment_status': patient['employment_status'],\n",
+    "                           'financial_stress': patient['financial_stress'],\n",
+    "                           'housing_insecurity': patient['housing_insecurity'],\n",
+    "                           'neighborhood_unsafety': patient['neighborhood_unsafety'],\n",
+    "                           'food_insecurity': patient['food_insecurity'],\n",
+    "                           'education_level': patient['education_level'],\n",
+    "                           'transportation_inaccessibility': patient['transportation_inaccessibility'],\n",
+    "                           'social_isolation': patient['social_isolation'],\n",
+    "                           'health_insurance_inadequacy': patient['health_insurance_inadequacy'],\n",
+    "                           'skipped_care_due_to_cost': patient['skipped_care_due_to_cost'],\n",
+    "                           'marital_status': patient['marital_status'],\n",
+    "                           'language_barrier': patient['language_barrier']})\n",
+    "\n",
+    "\n",
+    "sdoh_df = pd.DataFrame(flattened_data)\n",
+    "\n",
+    "sdoh_df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8394734d-7ff5-41ad-a008-a4b6b33dd1a9",
+   "metadata": {},
+   "source": [
+    "Nice! In just under 12 seconds we've parsed through five clinical notes, extracted discrete features and structured them into a neat table. That low latency is important for scaling up, and is why Groq's best in class speed makes it an ideal provider for this type of task - in an healthcare network with many providers, it would allow us to process clinical notes for 1500 patients in an hour."
```

Comment:
```suggestion
    "Nice! In just under 12 seconds we've parsed through five clinical notes, extracted discrete features and structured them into a neat table. That low latency is important for scaling up, and is why Groq's best-in-class speed makes it an ideal provider for this type of task - in a healthcare network with many providers, it would allow us to process clinical notes for 1500 patients in an hour."
```